### ISSUE TITLE
Translation support

### ISSUE THREAD

**Issue by mcccs** [OutCast3k/coinbin#56]
This website should allow multiple languages.

**Comment by dabura667** [comment]
I could handle Japanese.

**Comment by OutCast3k** [comment]
Seems like a good suggestion. Anyone else able or willing to translate?

**Comment by mcccs** [comment]
Yes, I'm sure that many users will translate, don't forget that this project has 80 forks.

**Comment by mcccs** [comment]
But the question is, how it'll work.
There are three options

- Transifex
- Crowdin
- Github Pull Request

**Comment by dabura667** [comment]
Crowdin +1

I am currently manager on the crowdin projects for breadwallet, Copay, and Telegram with Getgems.

I also wrote the script for Copay to build and update translations.

Transifex confuses me, and imo makes it hard to translate.

You can get a free repo if you are open source here:

https://crowdin.com/page/open-source-project-setup-request

**Comment by dabura667** [comment]
Then the question remains:

How should we handle translation?

1. Strings are sparsed across html and js files.
2. Angular gettext would be best suited, but this site doesn't use angular.
3. The sites need to be static.
4. We don't want the user to need to download all languages.

I'm thinking creating a dummy function for JS that will wrap all translatable strings like _("text")

Then for html, a "translate" attribute for the surrounding tag should work.

Then all we need to do is make a script that parses all the files and extracts them into a po format etc.

Then another script to replace those in the files and generate new html and js files in a subfolder.

Ghetto Angular gettext without Angular, basically.


Or just throw the html and js files into crowdin and hope their algorithms get it right.

**Comment by mcccs** [comment]
> Transifex confuses me, and imo makes it hard to translate.

Same, I prefer crowdin too.

**Comment by zilveer** [comment]
What happened with the translation suggestion ?
Any progress?
